src/promise.js

```diff
@@ -236,6 +236,7 @@ Promise.prototype._then = function (
                 receiver = target === this ? undefined : this._boundTo;
             }
         }
+        this._promiseChained(target);
```

Comment:
This should be the original, not `target`. `Target` is when a promise is following another promise:

```js
var b = new Promise(function() {});

var a = new Promise(function(resolve) {
    resolve(b); 
});

//a's target is b

// I chain from a, not b.
a.then(function() {
    
})

```